tests/runtime/aws_client_mock.c

```diff
@@ -0,0 +1,258 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
```

Comment:
I had to copy the aws_client_mock files to tests/runtime, because I didn't know how to include them from another folder here. It is a TODO, need to figure out how to achieve this without copying the files.